docs/async-await.pug

```diff
@@ -0,0 +1,192 @@
+
+            doc.save(function(err, updatedDoc) {
+                if(err) {
+                    handleError(err);
+                }
+
+                // Final logic is 2 callbacks deep
+                console.log(updatedDoc);
+            }); 
+        })
+    }
+
+    // Better.
+    function thenUpdate() {
+        MyModel.findOne({firstName: 'franklin', lastName: 'roosevelt'})
+            .then(function(doc) {
+                doc.middleName = 'delano';
+                return doc.save();
+            })
+            .then(console.log)
+            .catch(function(err) {
+                handleError(err);
+            });
+    };
+
+    // Best?
+    async function awaitUpdate() {
+        try {
+            const doc = await MyModel.findOne({
+              firstName: 'franklin',
+              lastName: 'roosevelt'
+            });
+
+            doc.middleName = 'delano';
+        
+            console.log(await doc.save());
+        }
+        
+        catch(err) {
+            handleError(err);
+        }
+    }
+    ```
+  :markdown
+    Note that the specific fulfillment values of different Mongoose methods vary, and may be affected by configuration. Please refer to the [API documentation](./api.html) for information about specific methods.
+
+  h3(id="async-functions") Async Functions
+  :markdown
+    Adding the keyword *async* to a JavaScript function automatically causes it to return a native JavaScript promise.  This is true [regardless of the return value we specify in the function body](http://thecodebarbarian.com/async-functions-in-javascript.html#an-async-function-always-returns-a-promise).     
+
+    ```javascript
+    async function getUser() {
+      //Inside getUser, we can await an async operation and interact with 
+      //foundUser as a normal, non-promise value...
+      const foundUser = await User.findOne({name: 'bill'});
+
+      console.log(foundUser); //Prints '{name: 'bill', admin: false}'
+      return foundUser;
+    }
+
+    //However, because async functions always return a promise, 
+    //user is a promise.
+    const user = getUser();
+
+    console.log(user)  //Oops.  Prints '[Promise]'
+    ```
+  :markdown
+    Instead, treat the return value of an async function as you would any other promise.  Await its fulfillment inside another async function, or chain onto it using ‘.then’ blocks.
+
+    ```javascript
+    async function getUser() {
+      const foundUser = await User.findOne({name: 'bill'});
+      return foundUser;
+    };
+
+    async function doStuffWithUser() {
+      //Await the promise returned from calling getUser.
+      const user = await getUser();
+
+      console.log(user);  //Prints '{name: 'bill', admin: false}'
+    }
+
+  h3(id="queries") Async/Await with Mongoose Queries
+  :markdown
+    Under the hood, [*async/await* is syntactic sugar](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await)
+<<<<<<< HEAD
```

Comment:
Looks like there's some merge conflicts in async-await.pug? Can you please clean those up? Otherwise, should be good to merge.

Reply:
This should be cleared up.  However, due to a botched git workflow, this PR has an issue with the Makefile and possibly other problems.  I opened a [PR for a new branch of this repo](https://github.com/Automattic/mongoose/pull/7969) that includes all suggested changes and builds correctly.  Should be a simple fast-forward onto master.  Sorry for the confusion.  Thanks!